internal/controllers/sync-volume/utils.go

```diff
@@ -41,7 +42,27 @@ import (
 )
 
 func hash(v storageos.Object) ([]byte, error) {
-	return json.Marshal(v)
+	stosVol, ok := v.GetInner().(stosapi.Volume)
+	if !ok {
+		return nil, errors.New("type cast error")
+	}
+
+	raw, err := json.Marshal(stosVol)
+	if err != nil {
+		return nil, err
+	}
+
+	hash := []byte{}
+	for _, s := range [][]byte{[]byte(stosVol.Id), []byte(stosVol.CreatedAt.String()), raw} {
```

Comment:
What's the benefit of hashing the `id`, the `createdAt` and the entire `raw` json string?

Doesn't the entire json string contain the `id` and `createdAt` anyway?

Reply:
The reason is we convert undefined number of any characters to ~19 digits. Because the output is slimmer than the input, we have some chance for collision. So this should power the output space and drastically decreases chance of collision.